onyo/lib/tests/test_commands_mv.py

```diff
@@ -239,3 +231,32 @@ def test_onyo_mv_and_rename(inventory: Inventory) -> None:
     assert inventory.repo.git.is_clean_worktree()
     assert not inventory.repo.is_inventory_dir(source)
     assert inventory.repo.is_inventory_dir(destination)
+
+
+@pytest.mark.ui({'yes': True})
+def test_onyo_mv_into_asset(inventory: Inventory) -> None:
+    asset = dict(some_key="some_value",
+                 type="TYPE",
+                 make="MAKE",
+                 model="MODEL",
+                 serial="SERIAL2",
+                 other=1,
+                 directory=inventory.root)
+    asset_path = inventory.root / "TYPE_MAKE_MODEL.SERIAL2"
+    inventory.add_asset(asset)
+    inventory.commit("Add second asset")
+
+    old_hexsha = inventory.repo.git.get_hexsha()
+    source = inventory.root / "somewhere" / "nested" / "TYPE_MAKER_MODEL.SERIAL"
+
+    onyo_mv(inventory,
+            source=source,
+            destination=asset_path,
+            message="Move asset into asset to create asset dir")
+    assert inventory.repo.git.get_hexsha('HEAD~1') == old_hexsha
+    assert inventory.repo.git.is_clean_worktree()
+    assert not source.exists()
+    # Turned destination into asset dir:
+    assert inventory.repo.is_asset_dir(asset_path)
```

Comment:
Nitpick: we might here want to test both, the caches and the filesystem, so as to be sure that both are updated through the change.

Reply:
That's not necessary here. We test the command did `commit`. Cache update only needs to be done when committing, so this is included in `commit` and tested for with the respective unit tests.

Reply:
Furthermore the `repo.is_xxx` methods are in fact using the cache.